app/custom-oauth/server/custom_oauth_server.js

```diff
@@ -308,50 +359,56 @@ export class CustomOAuth {
 			identity.name = this.getName(identity);
 		}
 
-		return identity;
+		return Object.entries(identity).reduce((result, [name, value]) => {
+			if (IDENTITY_PROPNAME_FILTER.test(name)) {
+				return result;
+			}
+			return {
+				...result,
+				[name]: value,
+			};
+		}, {});
 	}
 
 	retrieveCredential(credentialToken, credentialSecret) {
 		return OAuth.retrieveCredential(credentialToken, credentialSecret);
 	}
 
 	getUsername(data) {
-		let username = '';
+		const value = fromTemplate(this.usernameField, data);
 
-		username = this.usernameField.split('.').reduce(function(prev, curr) {
-			return prev ? prev[curr] : undefined;
-		}, data);
-
-		if (!username) {
+		if (!value) {
 			throw new Meteor.Error('field_not_found', `Username field "${ this.usernameField }" not found in data`, data);
 		}
-		return username;
+		return value;
 	}
 
-	getCustomName(data) {
-		let customName = '';
+	getEmail(data) {
+		const value = fromTemplate(this.emailField, data);
 
-		customName = this.nameField.split('.').reduce(function(prev, curr) {
-			return prev ? prev[curr] : undefined;
-		}, data);
+		if (!value) {
+			throw new Meteor.Error('field_not_found', `Email field "${ this.emailField }" not found in data`, data);
+		}
+		return value;
+	}
+
+	getCustomName(data) {
+		const value = fromTemplate(this.nameField, data);
 
-		if (!customName) {
+		if (!value) {
 			return this.getName(data);
 		}
 
-		return customName;
+		return value;
 	}
 
 	getAvatarUrl(data) {
-		const avatarUrl = this.avatarField.split('.').reduce(function(prev, curr) {
-			return prev ? prev[curr] : undefined;
-		}, data);
+		const value = fromTemplate(this.avatarField, data);
 
-		if (!avatarUrl) {
-			logger.debug(`Avatar field "${ this.avatarField }" not found in data`, data);
+		if (!value) {
+			throw new Meteor.Error('field_not_found', `Avatar field "${ this.avatarField }" not found in data`, data);
```

Comment:
this would be a breaking change... should potentially changed back to a simple `debug` log vs throwing an `error`.
let me know. happy to change